R/assign_datetime.R

```diff
@@ -0,0 +1,167 @@
+#'   element is taken as parsing format for each variable indicated in
+#'   `raw_var`. If a list is provided, then each element must be a character
+#'   vector of formats. The first vector of formats is used for parsing the
+#'   first variable in `raw_var`, and so on.
+#' @param tgt_var The target SDTM variable: a single string indicating the name
+#'   of variable to be derived.
+#' @param raw_unk A character vector of string literals to be regarded as
+#'   missing values during parsing.
+#' @param tgt_dat Target dataset: a data frame to be merged against `raw_dat` by
+#'   the variables indicated in `id_vars`. This parameter is optional, see
+#'   section Value for how the output changes depending on this argument value.
+#' @param id_vars Key variables to be used in the join between the raw dataset
+#'   (`raw_dat`) and the target data set (`raw_dat`).
+#' @param .warn Whether to warn about parsing failures.
+#'
+#' @returns The returned data set depends on the value of `tgt_dat`:
+#' - If no target dataset is supplied, meaning that `tgt_dat` defaults to
+#' `NULL`, then the returned data set is `raw_dat`, selected for the variables
+#' indicated in `id_vars`, and a new extra column: the derived variable, as
+#' indicated in `tgt_var`.
+#' - If the target dataset is provided, then it is merged with the raw data set
+#' `raw_dat` by the variables indicated in `id_vars`, with a new column: the
+#' derived variable, as indicated in `tgt_var`.
+#'
+#' @examples
+#' md1 <-
+#'   tibble::tribble(
+#'     ~oak_id, ~raw_source, ~patient_number, ~MDBDR,        ~MDEDR,        ~MDETM,
+#'     1L,      "MD1",       375,             NA,            NA,            NA,
+#'     2L,      "MD1",       375,             "15-Sep-20",   NA,            NA,
+#'     3L,      "MD1",       376,             "17-Feb-21",   "17-Feb-21",   NA,
+#'     4L,      "MD1",       377,             "4-Oct-20",    NA,            NA,
+#'     5L,      "MD1",       377,             "20-Jan-20",   "20-Jan-20",   "10:00:00",
+#'     6L,      "MD1",       377,             "UN-UNK-2019", "UN-UNK-2019", NA,
+#'     7L,      "MD1",       377,             "20-UNK-2019", "20-UNK-2019", NA,
+#'     8L,      "MD1",       378,             "UN-UNK-2020", "UN-UNK-2020", NA,
+#'     9L,      "MD1",       378,             "26-Jan-20",   "26-Jan-20",   "07:00:00",
+#'     10L,     "MD1",       378,             "28-Jan-20",   "1-Feb-20",    NA,
+#'     11L,     "MD1",       378,             "12-Feb-20",   "18-Feb-20",   NA,
+#'     12L,     "MD1",       379,             "10-UNK-2020", "20-UNK-2020", NA,
+#'     13L,     "MD1",       379,             NA,            NA,            NA,
+#'     14L,     "MD1",       379,             NA,            "17-Feb-20",   NA
+#'   )
+#'
+#' cm1 <-
+#'   assign_datetime(
+#'     raw_dat = md1,
+#'     raw_var = "MDBDR",
+#'     raw_fmt = "d-m-y",
+#'     raw_unk = c("UN", "UNK"),
+#'     tgt_var = "CMSTDTC"
+#'   )
+#'
+#' cm1
+#' problems(cm1$CMSTDTC)
+#'
```

Comment:
Can we add this example with date and time?

```R
cm1 <-
   assign_datetime(
     raw_dat = md1,
     raw_var = c("MDEDR","MDETM"),
     raw_fmt = c("d-m-y", "H:M:S"),
     raw_unk = c("UN", "UNK"),
     tgt_var = "CMSTDTC"
   )
```

Reply:
Sure! Will do.